File: Programming Basics/Nested Loops - Exercises/06.CinemaTickets.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        string movie = Console.ReadLine();

        double ticketsSold = 0, studentTickets = 0, standardTickets = 0, kidTickets = 0;

        while (movie != "Finish")
        {
            double seats = int.Parse(Console.ReadLine());
            double studentsCount = 0, standardsCount = 0, kidsCount = 0, totalTicketsCount = 0;

            for (int i = 0; i < seats; i++)
            {
                string type = Console.ReadLine();
                if (type == "End")
                {
                    break;
                }
                if (type == "student")
                {
                    studentsCount++;
                    studentTickets++;
                }
                else if (type == "standard")
                {
                    standardsCount++;
                    standardTickets++;
                }
                else if (type == "kid")
                {
                    kidsCount++;
                    kidTickets++;
                }
                totalTicketsCount++;
                ticketsSold++;
            }

            Console.WriteLine($"{movie} - {100 * totalTicketsCount / seats:f2}% full.");

            movie = Console.ReadLine();
        }

        Console.WriteLine($"Total tickets: {ticketsSold}");
        Console.WriteLine($"{100 * studentTickets / ticketsSold:f2}% student tickets.");
        Console.WriteLine($"{100 * standardTickets / ticketsSold:f2}% standard tickets.");
        Console.WriteLine($"{100 * kidTickets / ticketsSold:f2}% kids tickets.");
    }
}

File: Programming Basics - Python/Nested Loops - Exercises/06.CinemaTickets.py
student_tickets = 0
standard_tickets = 0
kid_tickets = 0
total_tickets_sold = 0

movie = input()

while movie != "Finish":
    capacity = int(input())
    taken_seats = 0

    while taken_seats < capacity:
        visitor_type = input()

        if visitor_type == "End":
            break

        if visitor_type == "student":
            student_tickets += 1
        elif visitor_type == "standard":
            standard_tickets += 1
        elif visitor_type == "kid":
            kid_tickets += 1

        taken_seats += 1
        total_tickets_sold += 1

    print(f"{movie} - {(taken_seats / capacity) * 100:.2f}% full.")
    movie = input()

print(f"Total tickets: {total_tickets_sold}")
print(f"{(student_tickets / total_tickets_sold) * 100:.2f}% student tickets.")
print(f"{(standard_tickets / total_tickets_sold) * 100:.2f}% standard tickets.")
print(f"{(kid_tickets / total_tickets_sold) * 100:.2f}% kids tickets.")